src/coreclr/jit/morph.cpp

```diff
@@ -8961,7 +8966,10 @@ GenTree* Compiler::fgMorphSmpOp(GenTree* tree, MorphAddrContext* mac, bool* optA
 
             GenTree* oldTree = tree;
 
-            tree = gtFoldExpr(tree);
+            if (opts.OptimizationEnabled())
+            {
+                tree = gtFoldExpr(tree);
```

Comment:
For the `gtFold...` seems like we should have a consistent pattern we use -- either the caller or the callee should check if optimizations are enabled, but not both?

Maybe this is covered in your opt levels PR?